### PR TITLE
[mig] Users ldap populate port v8 feat to v9

### PR DESCRIPTION
**Pull request by simahawk**
Twin PR #920 
/cc @hbrunn @gurneyalex @lasley

### CODE REVIEW

**Comment by lasley** [comment]
Dang looks like bots are 🔴

**Comment by simahawk** [comment]
d'oh! I lost `models/__init__.py` while fixing rebase conflicts :smile: Fixed!

**Comment by simahawk** [comment]
@gurneyalex rebased

File: users_ldap_populate/models/users_ldap.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                    _("No login attribute found: "
+                      "Could not extract login attribute from filter %s") %
+                    conf['ldap_filter'])
+            results = self.get_ldap_entry_dicts(conf)
+            for result in results:
+                login = result[1][login_attr][0].lower().strip()
+                user_id = self.with_context(
+                    no_reset_password=True
+                ).get_or_create_user(conf, login, result)
+                if not user_id:
```
Comment (p-tombez) [inline users_ldap_populate/models/users_ldap.py:76]: Couldn't this be done by using context option ```active_test=False``` and check if ```active``` needs to be changed ?

File: users_ldap_populate/models/users_ldap.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                        self.env['res.users'].sudo().browse(
+                            res[0]
+                        ).write(
+                            {'active': True}
+                        )
+                    else:
+                        raise UserError(
+                            _('Unable to process user with login %s' % login)
+                        )
+                known_user_ids.append(user_id)
```
Comment (p-tombez) [inline users_ldap_populate/models/users_ldap.py:94]: During the ```if``` block above, ```user_id``` is not reassigned, so you'll basically append a ```None``` in this list. Is it wanted ?

**Review by p-tombez — COMMENTED** [review]
LG, but I have some questions

**Comment by damdam-s** [comment]
tests issues seems not to be linked to this module

File: users_ldap_populate/models/users_ldap.py (hunk 1 above)
Comment (gurneyalex) [reply users_ldap_populate/models/users_ldap.py:76]: no because of the case-insensitive search on login

**Review by gurneyalex — APPROVED** [review]

**Review by tschanzt — APPROVED** [review]
Looks good to me skimming through. But not an expert on ldap

**Comment by moylop260** [comment]
Hi @simahawk 
Could you check the following warning, please?
[2019-01-30 10:46:09,002 6485 WARNING openerp_test openerp.models: res.company.ldap.create() includes unknown fields: ldap_port](https://travis-ci.org/OCA/server-tools/jobs/486346348#L5177)

The red ci seems unrelated, maybe rebasing are fixed. I think so.

Could you check it, please, in order to merge it

**Comment by damdam-s** [comment]
@moylop260 I take care of this one. Doing it right now

**Comment by moylop260** [comment]
The travis error is unrelated since that stable branch it red with the same errors:
 - https://travis-ci.org/OCA/server-tools/builds/486348852

**Pull request merged**